### PR TITLE
add initial powerbi assets

### PR DESCRIPTION
**Pull request by cnolanminich**
This PR adds Power BI assets to the Hooli project. Features include:

* Materializable semantic models
* Automatic integration with the lineage graph of an upstream dbt assets
* A semantic model with a downstream dashboard (see below for my amazing dashboarding skillz)
* Example in the code of using the translator, including adding tags and groups

image

Open Questions:

* Is this something we want to merge or just an example to keep around as a branch deployment?
* Is there a "flow" that tells the story better than "analysis of augmented orders"
* Should the group be "BI" or tied to some business area?

See below for a screenshot of what it looks like

image

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
             "requests_toolbelt",
             "html5lib",
             "scikit-learn",
+            "dagster-powerbi==0.0.11"
```
Comment (slopp) [inline setup.py:34]: do we want the pin forever? or should we let it float like the others post-merge?

File: hooli_data_eng/resources/__init__.py [hunk 2]
```diff
@@ -58,6 +59,7 @@
 
 resource_def = {
     "LOCAL": {
+        "power_bi": power_bi_workspace,
```
Comment (slopp) [inline hooli_data_eng/resources/__init__.py:62]: I wonder if there is something we can do here to mock it out so folks who clone and run this locally without the powerBI creds still work?

(Speaking as someone who will want to be able to run the demo locally without worrying if I have sourced the power BI creds correctly)

File: hooli_data_eng/powerbi_workspace.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from dagster import EnvVar
+from dagster_powerbi import PowerBIWorkspace, PowerBIServicePrincipal
+
+# Connect using a service principal
+power_bi_workspace = PowerBIWorkspace(
```
Comment (slopp) [inline hooli_data_eng/powerbi_workspace.py:5]: Can we throw these in 1pass and do a PSA to SE?

File: hooli_data_eng/definitions.py [hunk 4]
```diff
@@ -89,3 +89,6 @@
     ],
     jobs=[analytics_job, predict_job, dbt_slim_ci_job],
 )
+
+
+#defs = Definitions.merge(static_defs, powerbi_assets.build_defs(dagster_powerbi_translator=MyCustomPowerBITranslator, enable_refresh_semantic_models=True))
```
Comment (slopp) [inline hooli_data_eng/definitions.py:94]: glad we didn't need this! 

```suggestion
```

**Review by slopp — APPROVED** [review]
This is awesome! So excited to have it in, thanks for chasing all of this down

> Is this something we want to merge or just an example to keep around as a branch deployment?

Deploy

> Is there a "flow" that tells the story better than "analysis of augmented orders"
lgtm

> Should the group be "BI" or tied to some business area?
BI is my preference

I left a few other comments. My other request, if possible, can we add a link to the dashboard in the asset description? I don't want to have to remember how to get to it, and I think it adds some pizzaz to an otherwise empty asset card/overview page

**Comment by cnolanminich** [comment]
> This is awesome! So excited to have it in, thanks for chasing all of this down
> 
> > Is this something we want to merge or just an example to keep around as a branch deployment?
> 
> Deploy

Just to confirm, by "deploy" you mean merge into master post comment addressing?

File: setup.py (hunk 1 above)
Comment (cnolanminich) [reply setup.py:34]: I think we can remove now -- I was pinning aggressively due to major API shifts with the definintions loader that made it pretty dicey. That should all be resolved now

**Review by thomasganka — APPROVED** [review]
🐐

**Pull request merged**